waspc/packages/deploy/src/providers/fly/deploy/deploy.ts

```diff
@@ -0,0 +1,119 @@
+import { exit } from 'process';
+import { $, cd, fs } from 'zx';
+import {
+	cdToClientBuildDir,
+	cdToServerBuildDir,
+	displayWaspRocketImage,
+	makeIdempotent,
+	getCommandHelp,
+	waspSays,
+} from '../helpers/helpers.js';
+import {
+	clientTomlExistsInProject,
+	copyLocalClientTomlToProject,
+	copyLocalServerTomlToProject,
+	copyProjectClientTomlLocally,
+	copyProjectServerTomlLocally,
+	getInferredBasenameFromClientToml,
+	getInferredBasenameFromServerToml,
+	getTomlFilePaths,
+	serverTomlExistsInProject,
+} from '../helpers/tomlFileHelpers.js';
+import { DeployOptions } from './DeployOptions.js';
+import { createDeploymentInfo, DeploymentInfo } from '../DeploymentInfo.js';
+import { flySetupCommand } from '../index.js';
+import { secretExists } from '../helpers/flyctlHelpers.js';
+
+export async function deploy(options: DeployOptions): Promise<void> {
+	waspSays('Deploying your Wasp app to Fly.io!');
+
+	const buildWasp = makeIdempotent(async () => {
+		if (options.skipBuild) {
+			return;
+		}
+
+		waspSays('Building your Wasp app...');
+		cd(options.waspProjectDir);
+		await $`${options.waspExe} build`;
+	});
+
+	const tomlFilePaths = getTomlFilePaths(options);
+
+	// NOTE: Below, it would be nice if we could store the client, server, and DB names somewhere.
+	// For now we just rely on the suffix naming convention and infer from toml files.
+	if (!serverTomlExistsInProject(tomlFilePaths)) {
+		waspSays(`${tomlFilePaths.serverTomlPath} missing. Skipping server deploy. Perhaps you need to run "${getCommandHelp(flySetupCommand)}" first?`);
+	} else {
+		const inferredBaseName = getInferredBasenameFromServerToml(tomlFilePaths);
+		const deploymentInfo = createDeploymentInfo(inferredBaseName, undefined, options, tomlFilePaths);
+		await buildWasp();
+		await deployServer(deploymentInfo);
+	}
+
+	if (!clientTomlExistsInProject(tomlFilePaths)) {
+		waspSays(`${tomlFilePaths.clientTomlPath} missing. Skipping client deploy. Perhaps you need to run "${getCommandHelp(flySetupCommand)}" first?`);
+	} else {
+		const inferredBaseName = getInferredBasenameFromClientToml(tomlFilePaths);
+		const deploymentInfo = createDeploymentInfo(inferredBaseName, undefined, options, tomlFilePaths);
+		await buildWasp();
+		await deployClient(deploymentInfo);
+	}
+}
+
+async function deployServer(deploymentInfo: DeploymentInfo) {
+	waspSays('Deploying your server now...');
+
+	cdToServerBuildDir(deploymentInfo.options.waspProjectDir);
+	copyProjectServerTomlLocally(deploymentInfo.tomlFilePaths);
+
+	// Make sure we have a DATABASE_URL present. If not, they need to create/attach their DB first.
+	try {
+		const databaseUrlSet = await secretExists('DATABASE_URL');
+		if (!databaseUrlSet) {
+			waspSays('Your server app does not have a DATABASE_URL secret set. Perhaps you need to create or attach your database?');
```

Comment:
How can I do that as an user? Should we say something like "You can do that by running `wasp deploy fly db`"

Reply:
It could be the DB is missing, or that the DB is up but the secret was somehow not set (or deleted). So it is in a funky state and hard to give them exact directions here.